File: academic-system-java/src/main/java/org/example/academic/system/service/ReportService.java
package org.example.academic.system.service;

import org.example.academic.system.model.AcademicSystem;
import org.example.academic.system.report.ReportGenerator;
import org.example.academic.system.security.Session;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ReportService {

    private static final Logger logger = LoggerFactory.getLogger(ReportService.class);

    private final AcademicSystem academicSystem;

    // Injeta o modelo central para acessar a lista de turmas na geração dos relatórios
    public ReportService(AcademicSystem academicSystem) {
        this.academicSystem = academicSystem;
    }

    // Gera relatório listando todas as avaliações cadastradas por turma
    public String generateClassAssessmentSummaryReport() {
        logger.info("Generating class assessment summary report for role: {}", getCurrentUserRole());
        String report = ReportGenerator.classAssessmentSummaryReport(academicSystem.getClasses());
        logger.info("Class assessment summary report generated successfully");
        return report;
    }

    // Gera relatório mostrando o peso total das avaliações por turma e se está válido (soma = 1.0)
    public String generateAssessmentWeightReport() {
        logger.info("Generating assessment weight report for role: {}", getCurrentUserRole());
        String report = ReportGenerator.assessmentWeightReport(academicSystem.getClasses());
        logger.info("Assessment weight report generated successfully");
        return report;
    }

    // Obtém o papel do usuário autenticado na sessão para fins de log
    private String getCurrentUserRole() {
        var user = Session.getInstance().getAuthenticatedUser();
        return user != null ? user.getRole().toString() : "UNKNOWN";
    }
}

File: academic-system-python/academic_system/service/report_service.py
import logging

from ..model.academic_system import AcademicSystem
from ..report.report_generator import ReportGenerator
from ..security.session import Session

logger = logging.getLogger(__name__)


class ReportService:
    def __init__(self, academic_system: AcademicSystem):
        self._academic_system = academic_system

    def generate_class_assessment_summary_report(self) -> str:
        logger.info("Generating class assessment summary report for role: %s", self._current_role())
        report = ReportGenerator.class_assessment_summary_report(self._academic_system.get_classes())
        logger.info("Class assessment summary report generated successfully")
        return report

    def generate_assessment_weight_report(self) -> str:
        logger.info("Generating assessment weight report for role: %s", self._current_role())
        report = ReportGenerator.assessment_weight_report(self._academic_system.get_classes())
        logger.info("Assessment weight report generated successfully")
        return report

    def _current_role(self) -> str:
        user = Session.get_instance().get_authenticated_user()
        return str(user.role) if user else "UNKNOWN"
